helm/templates/deployments.yaml

```diff
@@ -0,0 +1,371 @@
+        - /gomplate
+        - --left-delim
+        - '%%'
+        - --right-delim
+        - '%%'
+        - --input-dir
+        - /sql-init-templates
+        - --output-dir
+        - /docker-entrypoint-initdb.d
+        image: hairyhenderson/gomplate:v3
+        name: gomplate
+        env:
+        - name: SERVERPASSWORD
+          {{- if .Values.server.storageCredentials.password }}
+          valueFrom:
+            secretKeyRef:
+              name: server-password
+              key: password
+              optional: false
+          {{- else }}
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.server.storageCredentials.passwordSecretName }}
+              key: {{ .Values.server.storageCredentials.passwordSecretKey }}
+              optional: false
+          {{- end }}
+        - name: SIGNERPASSWORD
+          {{- if .Values.signer.storageCredentials.password }}
+          valueFrom:
+            secretKeyRef:
+              name: signer-password
+              key: password
+              optional: false
+          {{- else }}
+          valueFrom:
+            secretKeyRef:
+              name: {{ .Values.signer.storageCredentials.passwordSecretName }}
+              key: {{ .Values.signer.storageCredentials.passwordSecretKey }}
+              optional: false
+          {{- end }}
+        volumeMounts:
+        - mountPath: /docker-entrypoint-initdb.d
+          name: sql-init
+        - mountPath: /sql-init-templates
+          name: notarysql
+      containers:
+      {{- if eq .Values.storage.flavor "mysql" }}
+      - args:
+        - mysqld
+        - --innodb_file_per_table
+        env:
+        - name: MYSQL_RANDOM_ROOT_PASSWORD
+          value: "yes"
+        image: {{ .Values.storage.image }}
+        name: mysql
+        ports:
+        - containerPort: 3306
+          name: mysql
+        volumeMounts:
+        - mountPath: /var/lib/mysql
+          name: notary-data
+      {{- else if eq .Values.storage.flavor "postgres" }}
+      - env:
+        - name: PGDATA
+          value: /var/lib/postgresql/data/pgdata
+        image: {{ .Values.storage.image }}
+        name: postgres
+        ports:
+        - containerPort: 5432
+          name: postgres
+        volumeMounts:
+        - mountPath: /var/lib/postgresql/data/pgdata
+          name: notary-data
+      {{- end }}
+        - mountPath: /docker-entrypoint-initdb.d
+          name: sql-init
+        - mountPath: /sql-init-templates
+          name: notarysql
+      volumes:
+      - name: notary-data
+        persistentVolumeClaim:
+          claimName: notary-data
+      - name: notarysql
+        configMap:
+          name: notarysql
+      - name: sql-init
+        emptyDir: {}
+
+---
+
+{{- end }}
+
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  labels:
+    app.kubernetes.io/name: notary
+    app.kubernetes.io/component: notary-server
+  name: notary-server
+spec:
+  replicas: {{ .Values.signer.replicas }}
```

Comment:
In notary-server component, you are assigning signer.replicas instead of server.replicas